configuration/src/chains/substrate/mod.rs

```diff
@@ -0,0 +1,54 @@
+//! Substrate tx submitter types
+
+use crate::agent::SignerConf;
+use std::str::FromStr;
+
+/// Substrate submitter type
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum SubmitterType {
+    /// Local sign/submit
+    Local,
+}
+
+impl FromStr for SubmitterType {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_ref() {
+            "local" => Ok(Self::Local),
+            _ => panic!("Unknown SubmitterType"),
+        }
+    }
+}
+
+/// Local or relay-based transaction submission
+#[derive(Debug, Clone, PartialEq, serde::Deserialize)]
+#[serde(tag = "submitterType", content = "submitter", rename_all = "camelCase")]
+pub enum TxSubmitterConf {
+    /// Signer configuration for local signer
+    Local(SignerConf),
+}
+
+impl From<crate::TxSubmitterConf> for TxSubmitterConf {
```

Comment:
Same comment as the one I made for `configuration/src/chains/ethereum/submitter/mod.rs`